query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala

```diff
@@ -199,8 +199,9 @@ class QueryResource(secondary: Secondary,
                                  headerSocrataLastModified -> schema.lastModified.toHttpDate) ++
             resourceName.map(fbf => Map(headerSocrataResource -> fbf)).getOrElse(Nil) ++
             (if (analyze) List("X-Socrata-Analyze" -> "true" ) else Nil)
+          val recvTimeout = queryTimeoutSeconds.map(x => Integer.parseInt(x) * 1000).getOrElse(receiveTimeout.toMillis.toInt)
```

Comment:
okay so, checking my understanding:
so before, our receiveTimeout was always being used here (as the timeout for query execution), which meant that:
* if query timeout was shorter than receiveTimeout, then the query could keep running even after connection was closed?
* if query timeout was longer than receiveTimeout, the user's request would fail/timeout despite them specifying a long timeout

And now we'll always use the queryTimeout if supplied.

Question:
Do we have any max value for what users can set queryTimeout to? seems a little dangerous if not

Reply:
The goals are
1. do not have query running longer than http timeout
2. honor request timeout > 10m if the request is debug (previously we can't exceed 10mins because of this http problem in QC which is fixed by this PR)
3. default timeout to 10mins if unspecified (previously - http is limited to 10 mins but the postgres side has no limit)

The above are achieved in a series of change in SF, QC and QS.

Reply:
thank you for the additional explanation!

Reply:
cool, yes I see the maximum setting in https://github.com/socrata-platform/soql-postgres-adapter/pull/459/files

What is a 'debug' query/are there restrictions on who can use it so that it won't cause load issues?

Reply:
debug is basically X-Socrata-Debug: true.  Only superadmin can set it to true.